src/transformers/models/detr/modeling_detr.py

```diff
@@ -1575,17 +1575,17 @@ def __init__(self, config: DetrConfig):
     @replace_return_docstrings(output_type=DetrSegmentationOutput, config_class=_CONFIG_FOR_DOC)
     def forward(
         self,
-        pixel_values,
-        pixel_mask=None,
-        decoder_attention_mask=None,
-        encoder_outputs=None,
-        inputs_embeds=None,
-        decoder_inputs_embeds=None,
-        labels=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        return_dict=None,
-    ):
+        pixel_values: torch.FloatTensor,
+        pixel_mask: Optional[torch.LongTensor] = None,
+        decoder_attention_mask: Optional[torch.LongTensor] = None,
```

Comment:
My IDE flagged that `decoder_attention_mask`, `inputs_embeds` & `decoder_inputs_embeds` are not used within the method. In the previous pr I realised about this but I left it just in case is some kind of backwards compatibility.

I've done some mini research on this and noticed that it feels like a common pattern for these detr models:
- `XForObjectDetection` (where `X` is the model, like `Detr`) does use those arguments but just because it's [calling the parent model](https://github.com/huggingface/transformers/blob/d8dfba59c5ca9a4898944e1f411302a6320d8b40/src/transformers/models/detr/modeling_detr.py#L1465) _to obtain encoder + decoder outputs_
- However, the parent model is not using them for anything.
- The models that contain the `ForSegmentation` variant are not using these arguments either.

As said, I kept them just in case it's a matter of backwards compatibility, let me know if you want me to remove them and, if yes, whether a different pr is needed.

Reply:
In type hint PRs, please leave the arguments there, even if they're unused! Ideally we'd like type hint PRs to be totally clean and to not affect the code at all except for adding hints.

Reply:
Fine, thanks for the answer :sparkles: